otherlibs/systhreads/multicore.ml

```diff
@@ -240,8 +248,12 @@ let rec manager_loop t =
       manager_loop t
     | _ ->
       let requests = Atomic.exchange t.incoming [] in
-      List.iter (fun (request : request) ->
-        (match Thread.Portable.create thread request.action with
+      List.iter (fun (Request request : request) ->
+        (match
+           Thread.Portable.create2
+             thread
+             #(request.action, request.argument)
```

Comment:
Hmm... Perhaps just pass the whole `request` to the `thread` function?
```ocaml
Thread.Portable.create thread request
```
No need to extend the `Thread` interface and should be efficient.